doc/languages-frameworks/maven.md

```diff
@@ -0,0 +1,191 @@
+{
+  "project": {
+    "artifactId": "mvn2nix-maven-plugin",
+    "groupId": "org.nixos.mvn2nix",
+    "version": "1.2.2",
+    "classifier": "",
+    "extension": "maven-plugin",
+    "dependencies": [
+      {
+        "artifactId": "javax.json-api",
+
+```
+
+This file is then given to the _buildMaven_ function, and it returns 3 properties.
+
+**repo**:
+    A Maven repository that is a symlink farm of all the dependencies found in the _project-info.json_
+
+
+**build**:
+    A simple derivation that runs through _mvn compile_ & _mvn package_ to build the JAR. You may use this as inspiration for more complicated derivations.
+
+Here is an example of building the Maven repository
+```nix
+{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+let build = buildMaven ./project-info.json;
+in
+build.repo
+```
+
+The benefit over the _double invocation_, is that the _/nix/store_ entry is a _linkFarm_ of every package, so that changes to your dependency set doesn't involve downloading everything from scratch.
+
+```bash
+❯ tree /nix/store/5zg4jxj7897lcasd7zpvjrk8drnmiaf5-maven-repository | head
+
+/nix/store/5zg4jxj7897lcasd7zpvjrk8drnmiaf5-maven-repository
+├── antlr
+│   └── antlr
+│       └── 2.7.2
+│           ├── antlr-2.7.2.jar -> /nix/store/d027c8f2cnmj5yrynpbq2s6wmc9cb559-antlr-2.7.2.jar
+│           └── antlr-2.7.2.pom -> /nix/store/mv42fc5gizl8h5g5vpywz1nfiynmzgp2-antlr-2.7.2.pom
+├── aopalliance
+│   └── aopalliance
+│       └── 1.0
+│           ├── aopalliance-1.0.jar -> /nix/store/yy66cplfasm9m5r61hi09vbk4mkk47rl-aopalliance-1.0.jar
+```
+
+## Runnable JAR
+
+You are able to run `mvn compile` but how do you get a runnable artifact without having to use something like building an UberJar ?
+
+We will use the same repositories we built above (either _double invocation_ or _buildMaven_) to setup a classpath for our Jar.
+
+Setup your _pom.xml_ to create a Jar with the following manifest:
+```xml
+<plugin>
+    <artifactId>maven-jar-plugin</artifactId>
+    <version>3.2.0</version>
+    <configuration>
+        <archive>
+            <manifest>
+                <addClasspath>true</addClasspath>
+                <classpathPrefix>lib/</classpathPrefix>
+                <classpathLayoutType>repository</classpathLayoutType>
+                <mainClass>$YOUR_MAIN_CLASS_HERE</mainClass>
+            </manifest>
+            <manifestEntries>
+                <Class-Path>.</Class-Path>
+            </manifestEntries>
+        </archive>
+    </configuration>
+</plugin>
+```
+
+> This instructs the Jar to look for the necessary dependencies in the _lib/_ relative folder. The layout of the folder is also in the _maven repository_ style.
+
+```nix
+# create a repository using one of the methods above
+let repository = ....;
+in
+stdenv.mkDerivation rec {
+  pname = "your-pkg";
+  version = "0.1";
+  name = "${pname}-${version}";
+  src = ./.;
+  buildInputs = [ jdk maven makeWrapper ];
+  buildPhase = ''
+    echo "Using repository ${repository}"
```

Comment:
```suggestion
```

Reply:
This is really not a bad idea to print, for easy troubleshooting when packaging. If `stdenv.mkDerivation` should set the bar, this logging is more useful than most of what's printed.

Reply:
I agree I kind of like the helpful echo here.